File: DevSupportAndroid/src/main/java/com/sup/dev/android/views/adapters/CardAdapter.java
package com.sup.dev.android.views.adapters;

import android.view.View;

import com.sup.dev.android.views.cards.Card;
import com.sup.dev.android.views.cards.CardSpoiler;
import com.sup.dev.android.views.cards.CardWidget;

public interface CardAdapter {

    View getView(Card card);

    void remove(Card card);

    int indexOf(Card card);

    boolean contains(Card card);

    void add(int i, Card card);
}

File: DevSupportAndroid/src/main/java/com/sup/dev/android/views/adapters/CardAdapter.kt
package com.sup.dev.android.views.adapters

import android.view.View
import com.sup.dev.android.views.cards.Card


interface CardAdapter {

    fun getView(card: Card): View?

    fun remove(card: Card)

    fun indexOf(card: Card): Int

    fun size(): Int

    operator fun get(i: Int): Card

    operator fun contains(card: Card): Boolean

    fun add(i: Int, card: Card)

    fun isVisible(card: Card): Boolean
}
